src/k8s-extension/azext_k8s_extension/partner_extensions/CostExport.py

```diff
@@ -0,0 +1,263 @@
+        msg = ("This operation requires requires registering the resource provider {0}. "
+               "We were unable to perform that registration on your behalf: "
+               "Server responded with error message -- {1} . "
+               "Please check with your admin on permissions, "
+               "or try running registration manually with: az provider register --wait --namespace {0}")
+        raise ValidationError(resource_provider, msg.format(e.args)) from e
+    logger.info("Resource provider '%s' is now registered.", resource_provider)
+
+
+def _is_resource_provider_registered(cmd, resource_provider, subscription_id=None):
+    registered = None
+    if not subscription_id:
+        subscription_id = get_subscription_id(cmd.cli_ctx)
+    try:
+        providers_client = _providers_client_factory(cmd.cli_ctx, subscription_id)
+        registration_state = getattr(providers_client.get(resource_provider), 'registration_state', "NotRegistered")
+
+        registered = (registration_state and registration_state.lower() == 'registered')
+    except Exception:  # pylint: disable=broad-except
+        pass
+    return registered
+
+
+def _providers_client_factory(cli_ctx, subscription_id=None):
+    return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES,
+                                   subscription_id=subscription_id).providers
+
+
+def _create_cost_export(cmd, subscription: str, mc_resource_group: str, cluster_name: str, storage_account_id: str,
+                        storage_container: str, storage_directory: str):
+    _register_resource_provider(cmd, "Microsoft.CostManagementExports")
+    args = [
+        "costmanagement", "export", "create",
+        "--name", cluster_name,
+        "--scope", f"/subscriptions/{subscription}/resourceGroups/{mc_resource_group}",
+        "--timeframe", "MonthToDate",
+        "--recurrence-period", f"from={datetime.datetime.now().strftime('%Y-%m-%dT%H:%M:%S')}",
+        "to=2200-01-01T00:00:00",
+        "--recurrence", "Daily",
+        "--schedule-status", "Active",
+        "--storage-account-id", storage_account_id,
+        "--storage-container", storage_container,
+        "--storage-directory", storage_directory,
+        "--query", "'id'",
+        "--subscription", subscription
+    ]
+    cli = _cli()
+    # Not every subscription can create AmortizedCost (prefered) export
+    # If it fails, try to create Usage export
+    cli.invoke(args + ["--type", "AmortizedCost"])
+    if cli.result.exit_code == 1 and isinstance(cli.result.error,
+                                                HttpResponseError) and cli.result.error.status_code == 400:
+        logger.info("couldn't create AmortizedCost export, trying Usage")
+        cli = _cli()
+        cli.invoke(args + ["--type", "Usage"])
+        if cli.result.exit_code != 0:
+            raise cli.result.error
+    elif cli.result.exit_code == 0:
+        logger.info("created cost export with 'Usage' type")
```

Comment:
This looks like it's indented incorrectly, or maybe the log line should be for '`AmortizedUsage'` type? If it's the latter, we probably want a log line confirming that the `'Usage'` one was created.

Reply:
ah good catch, I swapped the order but missed the log messages.

simplified the logic to avoid such errors